### PR TITLE
Fix dump rewrite result and dump extcall

### PR DESCRIPTION
**Pull request by FusionBolt**
1. support runtime dump extcall
2. dump_manager set append
3. dump Rewrite result with rule name

### CODE REVIEW

File: src/Native/src/kernels/stackvm/tensor_ops.cpp [hunk 1]
```diff
@@ -1032,6 +1017,19 @@ result<value_t> nncase::kernels::stackvm::fake_quantize(
...
+    [[maybe_unused]] value_t output, [[maybe_unused]] kernel_context &context) {
+    return err(std::errc::not_supported);
+}
+
+result<value_t> nncase::kernels::stackvm::bitcast(
+    [[maybe_unused]] prim_type_t type, [[maybe_unused]] prim_type_t new_type,
+    [[maybe_unused]] value_t input, [[maybe_unused]] value_t new_shape,
+    [[maybe_unused]] value_t output, [[maybe_unused]] kernel_context &context) {
+    return err(std::errc::not_supported);
+}
```
Comment (sunnycase) [inline src/Native/src/kernels/stackvm/tensor_ops.cpp:1032]: Why moving these code?

File: src/Nncase.Graph/Transform/RewriteProvider.cs [hunk 2]
```diff
@@ -3,6 +3,7 @@
 
 using System;
 using System.Collections.Generic;
+using System.Data;
```
Comment (sunnycase) [inline src/Nncase.Graph/Transform/RewriteProvider.cs:6]: using System.Data is not needed.

**Review by zhen8838 — DISMISSED** [review]

File: src/Nncase.Graph/Transform/RewriteProvider.cs [hunk 3]
```diff
@@ -33,6 +34,7 @@ public Expr Rewrite(Expr expr, IEnumerable<IRewriteRule> rules, RunPassContext c
                 post = visitor.Rewrite(last);
                 if (visitor.IsMutated)
                 {
+                    currentRule = rule;
```
Comment (zhen8838) [inline src/Nncase.Graph/Transform/RewriteProvider.cs:37]: Actual no need to use `currentRule`, when `IsMutated` the `lastRule` will be overridden by `rule`.

**Review by zhen8838 — DISMISSED** [review]

**Comment by sunnycase** [comment]
Use a meaningful title.

**Review by sunnycase — APPROVED** [review]

**Pull request merged**